sql/planbuilder/node_factory.go

```diff
@@ -1,29 +1,94 @@
 package planbuilder
 
 import (
-	"strings"
-
 	"github.com/dolthub/go-mysql-server/sql"
+	"github.com/dolthub/go-mysql-server/sql/expression"
 	"github.com/dolthub/go-mysql-server/sql/plan"
 	"github.com/dolthub/go-mysql-server/sql/transform"
 )
 
-func buildProject(p *plan.Project) sql.Node {
+// factory functions should apply all optimizations to an expression
+// that are always costing/simplification wins. Each function will be a series
+// of optimizations local to this specific node. Eventually there should be a
+// top level optimizer with a switch for every type.
+
+func factoryBuildProject(p *plan.Project) (sql.Node, error) {
```

Comment:
What is this name?

Reply:
you want me to just make it `factory.BuildProject` to start with? splitting it off into a package seemed premature